src/moneypenny/kubernetes.py

```diff
@@ -1,165 +1,315 @@
+        except ApiException as e:
+            logger.exception("Exception creating pod")
+            raise K8sApiException(e)
+        logger.debug(f"Pod for {username} created: {status}")
 
-    def _pod_status(self, name: str) -> str:
-        """Return the status of the pod with the given name.
+    def delete_objects(self, username: str) -> None:
+        """Delete both the pod and its associated configmap, given a
+        username.
 
         Parameters
         ----------
-        name: Name of the pod to check on.
-
-        Returns
-        -------
-        The pod status: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#podstatus-v1-core
+        name: Username for the pod and configmap to delete.
 
+        Raises
+        ------
+        K8sApiException if the deletion failed.
         """
-        try:
-            logger.info(f"Checking status for pod {name}")
-            pod = self.apps_api.read_namespaced_pod(
-                name, self.namespace
-            )
-        except ApiException as exc:
-            logger.exception("Exception reading pod")
-            raise K8sApiException(exc)
-        return pod.status
+        self._pod_delete(username)
+        self._configmap_delete(username)
 
-    def pod_completed(self, name: str) -> bool:
+    def check_pod_completed(self, username: str) -> bool:
         """Return true if the pod completed successfully, false if it
         is pending or running, and raise an exception if any part of the
         execution failed.
-        
+
         Parameters
         ----------
-        name: Name of the pod to check on.
+        name: Username for the pod to check on.
 
         Returns
         -------
         True for successful completion, False for still-in-progress.
 
         Raises
         ------
-        OperationFailed if the pod encountered an error.
+        PodNotFound if the pod isn't there at all, OperationFailed if the pod
+        encountered an error, K8sApiException for some other Kubernetes error.
         """
-
+        pname = _name_object(username, "pod")
         try:
-            status=self._pod_status(name)
+            pod = self.api.read_namespaced_pod(pname, self.namespace)
+            status = pod.status
         except ApiException as exc:
             if exc.status == 404:
-                raise PodNotFound(f"Pod {name} not found")
+                raise PodNotFound(f"Pod {pname} not found")
+            logger.exception("Error checking on pod completion")
             raise K8sApiException(exc)
-        phase = status.phase
+        phase: str = status.phase
         if phase == "Succeeded":
             return True
         if phase == "Pending" or "Running":
             return False
         if phase == "Unknown":
-            raise OperationFailed(f"Pod {name} in Unknown phase")
+            raise OperationFailed(f"Pod {pname} in Unknown phase")
         raise OperationFailed(status.message)
```

Comment:
This error seems a bit weird since it seems to be a catchall after 173?

Reply:
Clarified.  We reach that only if pod.status == "Failed" in which case we want to return the message as well since it may tell us why the pod failed.